[stable/postgresql] Allow priorityClass for pod priority and use with preemption

hobti01 opened the pull request:
**What this PR does / why we need it**:
Adds priorityClassName to the deployment, allowing prioritized scheduling.
https://kubernetes.io/docs/concepts/configuration/pod-priority-preemption/

Cleanup to remove unused strategy.  Strategy must always be Recreate and chart ignores any user-provided strategy property. https://github.com/helm/charts/pull/6343

**Which issue this PR fixes**: fixes #6955 


@databus23

cpanato commented on stable/postgresql/Chart.yaml:
```diff
@@ -1,5 +1,5 @@
 name: postgresql
-version: 0.17.0
+version: 0.17.1
```
since this is a new feature, I would say to bump to `0.18.0`

hobti01 replied:
Done

hobti01 committed:
Allow priorityClass

feat: allow setting of priorityClass in values
fix: remove duplicate strategy and unused property.  Strategy must 
always be Recreate https://github.com/helm/charts/pull/6343

Signed-off-by: Tim Hobbs <timothy.hobbs@ic-consult.com>

davidkarlsen commented on stable/postgresql/templates/deployment.yaml:
```diff
@@ -12,10 +12,6 @@ metadata:
 {{- end }}
 spec:
-{{- with .Values.strategy }}
-  strategy:
```
Won't this become rolling and not recreate due to default policy now? https://kubernetes.io/docs/concepts/workloads/controllers/deployment

hobti01 replied:
This was hard-coded to `Recreate` in 0.15.0 https://github.com/helm/charts/pull/6343

I'm simply removing the ignored option to set the policy.
